spec/controllers/users/two_factor_authentication_controller_spec.rb

```diff
@@ -572,13 +572,27 @@ def index
       end
 
       it 'rate limits confirmation OTPs on sign up' do
+        parsed_phone = Phonelib.parse(@unconfirmed_phone)
+        stub_analytics
         sign_in_before_2fa(@user)
         subject.user_session[:context] = 'confirmation'
         allow(IdentityConfig.store).to receive(:otp_delivery_blocklist_maxretry).and_return(999)
 
+        expect(@analytics).to receive(:track_event).with('OTP: Delivery Selection', anything).
```

Comment:
Can we use `have_logged_event` after the actions instead of `receive(:track_event)` so it uses all `FakeAnalytics` behaviors?

Reply:
yep, thanks, added [be2fd98](https://github.com/18F/identity-idp/pull/10360/commits/be2fd98096fb176697406f7e1560a88cb15ea183) to do that